internal/services/aws_reservation_service.go

```diff
@@ -52,19 +53,16 @@ func CreateAWSReservation(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Validate architecture match (hardcoded since image builder currently only supports x86_64). This can be only done
-	// when launch template is not set.
-	if payload.LaunchTemplateID == "" {
-		supportedArch := "x86_64"
-		it := preload.EC2InstanceType.FindInstanceType(clients.InstanceTypeName(payload.InstanceType))
-		if it == nil {
-			renderError(w, r, payloads.NewInvalidRequestError(r.Context(), fmt.Sprintf("unknown type: %s", payload.InstanceType), ErrUnknownInstanceTypeName))
-			return
-		}
-		if it.Architecture.String() != supportedArch {
-			renderError(w, r, payloads.NewWrongArchitectureUserError(r.Context(), ErrArchitectureMismatch))
-			return
-		}
+	// Validate architecture match.
+	supportedArch := "x86_64"
+	it := preload.EC2InstanceType.FindInstanceType(clients.InstanceTypeName(payload.InstanceType))
+	if it == nil {
+		renderError(w, r, payloads.NewInvalidRequestError(r.Context(), fmt.Sprintf("unknown type: %s", payload.InstanceType), ErrUnknownInstanceTypeName))
+		return
+	}
+	if it.Architecture.String() != supportedArch && it.Architecture.String() != "arm64" {
+		renderError(w, r, payloads.NewWrongArchitectureUserError(r.Context(), ErrArchitectureMismatch))
+		return
 	}
```

Comment:
Can you simply make `supportedArch` a slice and simplify the condition so we can easily add more later? This looks clunky now.

Reply:
Thank you, it was totally just my lazieness, thanks, fixed :+1: